docs/src/pages/docs/guides/ssr.md

```diff
@@ -5,59 +5,175 @@ title: SSR & Next.js
 
 - Static Generation (SSG)
 - Server-side Rendering (SSR)
 
-With React Query and Next.js, you can pre-render a page for SEO and gracefully upgrade that page's queries during hydration to support caching, invalidation and background refetching on the client side.
+React Query supports both of these forms of pre-rendering.
+
+## Prefetch the data yourself and pass it in as `initialData`
 
-For example, together with Next.js's [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation), you can pass the pre-fetched data for the page to `useQuery`'s' `initialData` option:
+Together with Next.js's [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation), you can pass the pre-fetched data for the page to `useQuery`'s' `initialData` option:
 
 ```jsx
 export async function getStaticProps() {
   const posts = await getPosts()
   return { props: { posts } }
 }
 
-function Props(props) {
+function Posts(props) {
   const { data } = useQuery('posts', getPosts, { initialData: props.posts })
 
   // ...
 }
 ```
 
-This page would be prerendered using the data fetched in `getStaticProps` and be ready for SEO, then, as soon it mounts on the client, will also be cached and refetched/updated in the background as needed.
+The setup is minimal and this can be a perfect solution for some cases, but there are a few tradeoffs compared to the full approach:
+
+* If you are calling `useQuery` in a component deeper down in the tree you need to pass the `initialData` down to that point
+* If you are calling `useQuery` with the same query in multiple locations, you need to pass `initialData` to all of them
+* There is no way to know at what time the query was fetched on the server, so `updatedAt` and determining if the query needs refetching is based on when the page loaded instead
+
+## Prefetch the query via React Query and use de/rehydration
 
-## Advanced SSR Concepts
+React Query supports prefetching a query on the server and handing off or _dehydrating_ that query to the client. This means the server can prerender markup that is immediately available on page load and as soon as JS is available, React Query can upgrade or _hydrate_ those queries with the full functionality of the library. This includes refetching those queries on the client if they have become stale since the time they were rendered on the server.
 
-When using SSR (server-side-rendering) with React Query there are a few things to note:
+### Integrating with Next.js
+
+To support caching queries on the server and set up hydration, you start with wrapping your application with `<ReactQueryCacheProvider>` in `_app.js`.
+
+> Note: You need to import `ReactQueryCacheProvider` from `'react-query/hydration'` for it to support hydration!
+
+```jsx
+// _app.jsx
+import { ReactQueryCacheProvider } from 'react-query/hydration'
+
```

Comment:
Might be good to explicitly create the query cache here as the default one is going to be removed.

Reply:
My understanding is that the global cache is being removed, but that `ReactQueryCacheProvider` will still create a cache for you by default if you don't provide one yourself?

Reply:
Ah I'm actually not sure, maybe it was just my assumption it would become more explicit like with `ApolloProvider`. But maybe we should keep it like this then